bin/node/src/main.rs

```diff
@@ -122,3 +122,52 @@ async fn main() -> anyhow::Result<()> {
         },
     }
 }
+
+/// Generates the `--version` output enriched with additional build metadata.
+fn long_version() -> String {
+    let version = env!("CARGO_PKG_VERSION");
+
+    let mut branch = env!("VERGEN_GIT_BRANCH");
+    let mut sha = env!("VERGEN_GIT_SHA").to_owned();
+    let dirty = env!("VERGEN_GIT_DIRTY");
+    if dirty == "true" {
+        sha = format!("{sha}-dirty");
+    }
+
+    // This is the default value set by `vergen` when these values are missing.
+    // The git values can be missing for a published crate, and while we do attempt
+    // to set default values in the build.rs, its still possible for these to be skipped
+    // e.g. when cargo publish --allow-dirty is used.
+    if branch == "VERGEN_IDEMPOTENT_OUTPUT" {
+        branch = "";
+    }
+    if sha == "VERGEN_IDEMPOTENT_OUTPUT" {
+        sha.clear();
+    }
+
+    let rust_version = env!("VERGEN_RUSTC_SEMVER");
+    let host = env!("VERGEN_RUSTC_HOST_TRIPLE");
+    let target = env!("VERGEN_CARGO_TARGET_TRIPLE");
+
+    let mut features = env!("VERGEN_CARGO_FEATURES");
+    if features.is_empty() {
+        features = "None";
+    }
+
+    let opt_level = env!("VERGEN_CARGO_OPT_LEVEL").to_owned();
+    let debug = env!("VERGEN_CARGO_DEBUG");
+
+    format!(
+        "{version}
+
+SHA:          {sha}
+branch:       {branch}
+features:     {features}
+rust version: {rust_version}
+target arch:  {target}
+host arch:    {host}
+opt-level:    {opt_level}
+debug:        {debug}
+"
```

Comment:
Maybe we should create a helper struct in the utils crate - something like `LongVersion` - that will encapsulate some of this logic? We could then still use `env!` macros here like:
```rust
let long_version = LongVersion {
    version: env!("CARGO_PKG_VERSION"),
    branch: env!("VERGEN_GIT_BRANCH"),
    sha: env!("VERGEN_GIT_SHA"),
    rust_version: env!("VERGEN_RUSTC_SEMVER"),
    host: env!("VERGEN_RUSTC_HOST_TRIPLE"),
    target: env!("VERGEN_CARGO_TARGET_TRIPLE"),
    features: env!("VERGEN_CARGO_FEATURES"),
    opt_level: env!("VERGEN_CARGO_OPT_LEVEL"),
    debug: env!("VERGEN_CARGO_DEBUG"),
};

long_version.to_string()
```

Reply:
I moved everything instead :)